tools/brot.cc

```diff
@@ -0,0 +1,393 @@
+\n\
+Licensed under the Apache License, Version 2.0 (the \"License\");\n\
+you may not use this file except in compliance with the License.\n\
+You may obtain a copy of the License at\n\
+\n\
+http://www.apache.org/licenses/LICENSE-2.0\n\
+\n");
+}
+
+void show_version()
+{
+  printf("brotli version " VERSION "\n");
+}
+
+void show_usage(const char* progname)
+{
+  show_version();
+
+  printf("\
+usage %s [flags] [input files...]\n\
+\n\
+  -c, --stdout      compress to stdout\n\
+  -d, --decompress  decompress\n\
+  -f, --force       force overwrite of output files, writing to terminal\n\
+  -k, --keep        keep original input files\n\
+  -L, --license     show license\n\
+  -q, --quiet       be quiet\n\
+  -v, --verbose     be verbose\n\
+  -V, --version     show version information\n\
+  -t, --test        test input archive\n\
+  -1, --fast        fastest compression\n\
+  -9, --best        best compression\n\
+\n\
+If no FILE or when FILE is -, read standard input.\n\
+\n\
+Report bugs to " BUGREPORT "\n", progname);
+}
+
+/* Prints a helpful message and exit with error code 1 */
+void exit_emit_try_help()
+{
+  errx(1,"Try '%s --help' for more information.",progname);
+}
+
+/* Parse command line options and input file names.
+   Sets global variable according to options.
+   Sets the 'file_list' and 'file_count' according to non-option parameters.
+   If no files given, sets 'file_list' to 'stdin_file_list'.  */
+void parse_command_line(int argc, char** argv)
+{
+  const char *progname = basename(argv[0]);
+
+  int ch;
+  static struct option longopts[] = {
+    {"best",      no_argument,   NULL,    '9'},
+    {"decompress",no_argument,   NULL,    'd'},
+    {"fast",      no_argument,   NULL,    '1'},
+    {"force",     no_argument,   NULL,    'f'},
+    {"help",      no_argument,   NULL,    'h'},
+    {"keep",      no_argument,   NULL,    'k'},
+    {"license",   no_argument,   NULL,    'L'},
+    {"quiet",     no_argument,   NULL,    'q'},
+    {"stdout",    no_argument,   NULL,    'c'},
+    {"test",      no_argument,   NULL,    't'},
+    {"verbose",   no_argument,   NULL,    'v'},
+    {"version",   no_argument,   NULL,    'V'},
+    {NULL,        0,             NULL,    0}
+  };
+
+  /* Change operation based on program name */
+  if (strcmp(progname,DECOMP_PROGNAME)==0)
+    opmode = DECOMPRESS;
+  if (strcmp(progname,CAT_PROGNAME)==0) {
+    to_stdout = 1;
+    opmode = DECOMPRESS;
+  }
+
+  while ((ch = getopt_long(argc, argv,
+                           "123456789cdfhkLqtvV", longopts, NULL)) != -1)
+    switch (ch) {
+      case '1':
+      case '2':
+      case '3':
+      case '4':
+      case '5':
+      case '6':
+      case '7':
+      case '8':
+      case '9':
+        /* TODO: adapt quality to Brotli scale */
+        compression_quality = (ch - '9');
+        break;
+
+      case 'c':
+        opmode = COMPRESS;
+        to_stdout = 1;
+        break;
+
+      case 'd':
+        opmode = DECOMPRESS;
```

Comment:
May be allow only one of `c` or `d`.

Let the default value of `opmode` be `UNKNOWN` with:

```c
typedef struct _opmode {
   UNKNOWN,
   COMPRESS,
   DECOMPRESS
} opmode;
```